python/lsst/daf/butler/registry/queries/expressions/_predicate.py

```diff
@@ -241,6 +247,23 @@ def visitBinaryOp(
             ]:
                 rhs = ColumnLiteral(self.to_datetime(rhs.value), datetime.datetime)
                 return lhs.predicate_method(self.OPERATOR_MAP[operator], rhs)
+            # Allow comparisons between astropy.time.Time expressions and
+            # datetime literals/binds, by coercing the
+            # datetime literals to astropy.time.Time (in UTC scale).
```

Comment:
Someday I'd like to deprecate this new code and some of the old code around here and say we only support `astropy.time.Time`. I'm not in a rush to do that now, but we should keep it in mind as we make RFCs to transition the usage of TIMESTAMP in the schema.

Reply:
I added this only to keep existing unit tests happy, It should indeed disappear once we remove datetime support.